### PR TITLE
Various mappings in net/minecraft/client/multiplayer

### PR DESCRIPTION
**Pull request by PiTheGuy**

### CODE REVIEW

File: data/net/minecraft/client/multiplayer/ServerData.mapping [hunk 1]
```diff
@@ -7,19 +7,31 @@ CLASS net/minecraft/client/multiplayer/ServerData
...
 		COMMENT Returns the base-64 encoded representation of the server's icon, or null if not available
 	METHOD isLan ()Z
 		COMMENT Return true if the server is a LAN server
 	METHOD read (Lnet/minecraft/nbt/CompoundTag;)Lnet/minecraft/client/multiplayer/ServerData;
 		COMMENT Takes an NBTTagCompound with 'name' and 'ip' keys, returns a ServerData instance.
 		ARG 0 nbtCompound
+	METHOD setChatPreviewEnabled (Z)V
+		ARG 1 chatPreviewEnabled
+	METHOD setEnforcesSecureChat (Z)V
+		ARG 1 enforcedSecureChat
```
Comment (sciwhiz12) [inline data/net/minecraft/client/multiplayer/ServerData.mapping]: Parameter to field assignment follows field name.
```suggestion
		ARG 1 enforcesSecureChat
```

File: data/net/minecraft/client/multiplayer/chat/ChatListener.mapping [hunk 2]
```diff
@@ -1,5 +1,64 @@
...
+		ARG 3 playerInfo
+		ARG 4 timestamp
+	METHOD guessChatUUID (Lnet/minecraft/network/chat/Component;)Ljava/util/UUID;
+		ARG 1 message
+	METHOD handleChatHeader (Lnet/minecraft/network/chat/SignedMessageHeader;Lnet/minecraft/network/chat/MessageSignature;[B)V
+		ARG 1 header
+		ARG 2 signature
+	METHOD handleChatMessage (Lnet/minecraft/network/chat/PlayerChatMessage;Lnet/minecraft/network/chat/ChatType$Bound;)V
+		ARG 1 chatMessage
+		ARG 2 bound
```
Comment (sciwhiz12) [inline data/net/minecraft/client/multiplayer/chat/ChatListener.mapping]: Might be better to have this be `boundChatType`.
```suggestion
		ARG 2 boundChatType
```

File: data/net/minecraft/client/multiplayer/chat/ChatListener.mapping [hunk 3]
```diff
@@ -1,5 +1,64 @@
...
+		ARG 2 isOverlay
+	METHOD isSenderLocalPlayer (Ljava/util/UUID;)Z
+		ARG 1 sender
+	METHOD logPlayerHeader (Lnet/minecraft/network/chat/SignedMessageHeader;Lnet/minecraft/network/chat/MessageSignature;[B)V
+		ARG 1 header
+		ARG 2 headSignature
+		ARG 3 bodyDigest
+	METHOD logPlayerMessage (Lnet/minecraft/network/chat/PlayerChatMessage;Lnet/minecraft/network/chat/ChatType$Bound;Lnet/minecraft/client/multiplayer/PlayerInfo;Lnet/minecraft/client/multiplayer/chat/ChatTrustLevel;)V
+		ARG 1 message
+		ARG 2 bound
```
Comment (sciwhiz12) [inline data/net/minecraft/client/multiplayer/chat/ChatListener.mapping]: Might be better to have this be `boundChatType`.
```suggestion
		ARG 2 boundChatType
```

File: data/net/minecraft/client/multiplayer/chat/ChatListener.mapping [hunk 4]
```diff
@@ -1,5 +1,64 @@
...
+		ARG 3 playerInfo
+		ARG 4 trustLevel
+	METHOD logSystemMessage (Lnet/minecraft/network/chat/Component;Ljava/time/Instant;)V
+		ARG 1 message
+		ARG 2 timeStamp
+	METHOD processNonPlayerChatMessage (Lnet/minecraft/network/chat/ChatType$Bound;Lnet/minecraft/network/chat/PlayerChatMessage;Lnet/minecraft/network/chat/Component;)Z
+		ARG 1 bound
```
Comment (sciwhiz12) [inline data/net/minecraft/client/multiplayer/chat/ChatListener.mapping]: Might be better to have this be `boundChatType`.
```suggestion
		ARG 1 boundChatType
```

File: data/net/minecraft/client/multiplayer/chat/ChatListener.mapping [hunk 5]
```diff
@@ -1,5 +1,64 @@
...
+		ARG 2 headSignature
+		ARG 3 bodyDigest
+	METHOD logPlayerMessage (Lnet/minecraft/network/chat/PlayerChatMessage;Lnet/minecraft/network/chat/ChatType$Bound;Lnet/minecraft/client/multiplayer/PlayerInfo;Lnet/minecraft/client/multiplayer/chat/ChatTrustLevel;)V
+		ARG 1 message
+		ARG 2 bound
+		ARG 3 playerInfo
+		ARG 4 trustLevel
+	METHOD logSystemMessage (Lnet/minecraft/network/chat/Component;Ljava/time/Instant;)V
+		ARG 1 message
+		ARG 2 timeStamp
```
Comment (sciwhiz12) [inline data/net/minecraft/client/multiplayer/chat/ChatListener.mapping]:
```suggestion
		ARG 2 timestamp
```

File: data/net/minecraft/client/multiplayer/chat/ChatListener.mapping [hunk 6]
```diff
@@ -1,5 +1,64 @@
...
+	METHOD processNonPlayerChatMessage (Lnet/minecraft/network/chat/ChatType$Bound;Lnet/minecraft/network/chat/PlayerChatMessage;Lnet/minecraft/network/chat/Component;)Z
+		ARG 1 bound
+		ARG 2 chatMessage
+		ARG 3 message
+	METHOD processPlayerChatHeader (Lnet/minecraft/network/chat/SignedMessageHeader;Lnet/minecraft/network/chat/MessageSignature;[B)Z
+		ARG 1 header
+		ARG 2 headerSignature
+		ARG 3 bodyDigest
+	METHOD processPlayerChatMessage (Lnet/minecraft/network/chat/ChatType$Bound;Lnet/minecraft/network/chat/PlayerChatMessage;Lnet/minecraft/network/chat/Component;Lnet/minecraft/client/multiplayer/PlayerInfo;ZLjava/time/Instant;)Z
+		ARG 1 bound
```
Comment (sciwhiz12) [inline data/net/minecraft/client/multiplayer/chat/ChatListener.mapping]: Might be better to have this be `boundChatType`.
```suggestion
		ARG 1 boundChatType
```

File: data/net/minecraft/client/multiplayer/chat/ChatListener.mapping [hunk 7]
```diff
@@ -1,5 +1,64 @@
...
+		ARG 4 playerInfo
+		ARG 6 timestamp
+	METHOD removeFromDelayedMessageQueue (Lnet/minecraft/network/chat/MessageSignature;)Z
+		ARG 1 signature
+	METHOD resolveSenderPlayer (Ljava/util/UUID;)Lnet/minecraft/client/multiplayer/PlayerInfo;
+		ARG 1 sender
 	METHOD setMessageDelay (D)V
 		ARG 1 delaySeconds
+	METHOD showMessageToPlayer (Lnet/minecraft/network/chat/ChatType$Bound;Lnet/minecraft/network/chat/PlayerChatMessage;Lnet/minecraft/network/chat/Component;Lnet/minecraft/client/multiplayer/PlayerInfo;ZLjava/time/Instant;)Z
+		ARG 1 bound
```
Comment (sciwhiz12) [inline data/net/minecraft/client/multiplayer/chat/ChatListener.mapping]: Might be better to have this be `boundChatType`.
```suggestion
		ARG 1 boundChatType
```

File: data/net/minecraft/client/multiplayer/chat/LoggedChatMessage.mapping [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+	METHOD canReport (Ljava/util/UUID;)Z
+		ARG 1 uuid
+	METHOD player (Lcom/mojang/authlib/GameProfile;Lnet/minecraft/network/chat/Component;Lnet/minecraft/network/chat/PlayerChatMessage;Lnet/minecraft/client/multiplayer/chat/ChatTrustLevel;)Lnet/minecraft/client/multiplayer/chat/LoggedChatMessage$Player;
+		ARG 0 profile
+		ARG 1 displayName
+		ARG 2 message
+		ARG 3 trustLevel
+	METHOD system (Lnet/minecraft/network/chat/Component;Ljava/time/Instant;)Lnet/minecraft/client/multiplayer/chat/LoggedChatMessage$System;
+		ARG 0 message
+		ARG 1 timeStamp
```
Comment (sciwhiz12) [inline data/net/minecraft/client/multiplayer/chat/LoggedChatMessage.mapping]:
```suggestion
		ARG 1 timestamp
```

File: data/net/minecraft/client/multiplayer/chat/report/AbuseReportSender.mapping [hunk 9]
```diff
@@ -0,0 +1,4 @@
+CLASS net/minecraft/client/multiplayer/chat/report/AbuseReportSender
+	METHOD create (Lnet/minecraft/client/multiplayer/chat/report/ReportEnvironment;Lcom/mojang/authlib/minecraft/UserApiService;)Lnet/minecraft/client/multiplayer/chat/report/AbuseReportSender;
+		ARG 0 enviroment
```
Comment (sciwhiz12) [inline data/net/minecraft/client/multiplayer/chat/report/AbuseReportSender.mapping]: Typo:
```suggestion
		ARG 0 environment
```

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]

File: data/net/minecraft/client/multiplayer/ServerData.mapping (hunk 1 above)
Comment (PiTheGuy) [reply data/net/minecraft/client/multiplayer/ServerData.mapping]: I think this is a typo.

**Comment by PiTheGuy** [comment]
Everything was committed

**Review by sciwhiz12 — APPROVED** [review]

**Pull request merged**